### PR TITLE
fixes an idle test and some text

### LINKED ISSUE #1974: Waypoint: Use Conditional Logic with IfElse Statements fails at random
The first test case fails at random, even with a right solution. Reported in Gitter

### LINKED ISSUE #2803: Bad condition for succeed - bug
Challenge http://www.freecodecamp.com/challenges/waypoint-use-conditional-logic-with-ifelse-statements has an issue. Please describe how to reproduce it, and include links to screenshots if possible.

"Create an if-else statement to return heads if the flip var is zero, or else return tails if it's not."

The condition is exactly opposite. If the flip var is zero, it has to return "tails" for the test to succeed.

flip-tails-heads

### PR DESCRIPTION
**Pull request by bugron**
Closes #2803.
closes #1974 

I know that `assert(editor.getValue().match(/if/g).length >= 5` looks ugly (because of JSON.string**if**y) but I wanted our campers to see to results of the test so I used `JSON.stringify`. I could use `toString()` but what's done is done and as @QuincyLarson says "Done is better than perfect".

### CODE REVIEW

**Comment by ahstro** [comment]
If `toString()` could be used, I'm pretty sure it should be since it's more descriptive and ubiquitous. If `y` is never going to be an object or nested array, please check out `git commit --amend`. :)

**Comment by bugron** [comment]
@ahstro done. I actually used `JSON.stringify` because a lot of challenges are designed to use it. Anyway I replaced it with `toString()`.

File: seed/challenges/basic-javascript.json [hunk 1]
```diff
@@ -974,13 +974,13 @@
         "Create an <code>if-else statement</code> to return <code>heads</code> if the flip var is zero, or else return <code>tails</code> if it's not."
       ],
       "tests":[
-        "assert((function(){if(myFunction() === \"heads\" || myFunction() === \"tails\"){return true;}else{return false;}})(), 'myFunction should either return heads or tails');",
-        "assert(editor.getValue().match(/if/g).length >= 3, 'You should have created a new if statement');",
-        "assert(editor.getValue().match(/else/g).length >= 2, 'You should have created a new else statement');"
+        "assert((function(){var result = myFunction();if((result === \"heads\" && flip === 0) || (result === \"tails\" && flip !== 0)){return true;}else{return false;}})(), '<code>myFunction</code> should either return <code>heads</code> or <code>tails</code> if all conditions met.');",
```
Comment (benschenker) [inline seed/challenges/basic-javascript.json]: I'd recommend splitting this out to be two tests (like [here](https://github.com/FreeCodeCamp/FreeCodeCamp/pull/2158#issuecomment-138386354)):
1. one that tests that the result is either `heads` or `tails` - `myFunction should either return heads or tails`
2. one that tests if flip is the correct value - `myFunction should return heads when flip equals 0 and tails when flip equals 1`

**Comment by bugron** [comment]
@benschenker done. Is it better now?

File: seed/challenges/basic-javascript.json [hunk 2]
```diff
@@ -974,13 +974,14 @@
...
-        "assert(editor.getValue().match(/if/g).length >= 3, 'You should have created a new if statement');",
-        "assert(editor.getValue().match(/else/g).length >= 2, 'You should have created a new else statement');"
+        "assert((function(){var result = myFunction();if(result === 'heads' || result === 'tails'){return true;} else {return false;}})(), '<code>myFunction</code> should either return <code>heads</code> or <code>tails</code>.');",
+        "assert((function(){var result = myFunction();if(result === 'heads' && flip === 0 || result === 'tails' && flip !== 0){return true;} else {return false;}})(), '<code>myFunction</code> should return <code>heads</code> when flip equals 0 and <code>tails</code> when flip equals 1.');",
+        "assert(editor.getValue().match(/if/g).length >= 4, 'You should have created a new if statement.');",
+        "assert(editor.getValue().match(/else/g).length >= 2, 'You should have created a new else statement.');"
       ],
       "challengeSeed":[
+        "var flip = Math.floor(Math.random() * (1 - 0 + 1)) + 0;",
```
Comment (benschenker) [inline seed/challenges/basic-javascript.json]: Why did you move this line up here?

Comment (bugron) [inline seed/challenges/basic-javascript.json]: Because `flip` is in `myFunction()`'s scope.

**Comment by benschenker** [comment]
Also @ahstro's test code [here](https://github.com/FreeCodeCamp/FreeCodeCamp/pull/2158#issuecomment-138386354) is a bit more readable, not a huge deal though.

File: seed/challenges/basic-javascript.json (hunk 2 above)
Comment (benschenker) [inline seed/challenges/basic-javascript.json]: This will make flip have one value for the all the tests, usually we might want it to run with a different value for each test but it may work here.

**Comment by bugron** [comment]
@benschenker I agree, but in all challenges you can find that part of code unformatted. After merging/closing this PR one can refactor all challenges.

**Comment by benschenker** [comment]
Looks good to me then, haven't tested it locally though.  Have you?

File: seed/challenges/basic-javascript.json (hunk 2 above)
Comment (bugron) [inline seed/challenges/basic-javascript.json]: @benschenker you can try in your local that `flip` gets different value.

**Comment by bugron** [comment]
@benschenker absolutely. Maybe @ahstro would like to test it in local?

File: seed/challenges/basic-javascript.json (hunk 2 above)
Comment (benschenker) [inline seed/challenges/basic-javascript.json]: Don't have it up and running locally yet, but I trust you!

Comment (bugron) [inline seed/challenges/basic-javascript.json]: @benschenker thank you but I insist that someone else needs to check this in local.

**Comment by ahstro** [comment]
I'm pari programming in a minute, so I can't check it right now, but I'll definitely read through and try it out later tonight.

**Comment by ahstro** [comment]
Sorry for the delay, got caught up in some other stuff. It's working fine locally, but what happened to removing the padding code?

**Comment by bugron** [comment]
@ahstro did you mean [this](https://github.com/FreeCodeCamp/FreeCodeCamp/pull/2921#issuecomment-138601155)?

**Comment by ahstro** [comment]
@bugron Oh, yeah, didn't get that at first. Well, in that case, this is working perfectly.
